corehq/form_processor/submission_post.py

```diff
@@ -219,14 +212,15 @@ def run(self):
         report_submission_usage(self.domain)
         failure_response = self._handle_basic_failure_modes()
         if failure_response:
-            return FormProcessingResult(failure_response, None, [], [], 'known_failures')
+            return FormProcessingResponse(failure_response, None, [], [], 'known_failures')
 
-        result = process_xform_xml(self.domain, self.instance, self.attachments, self.auth_context.to_json())
-        submitted_form = result.submitted_form
-
-        self._post_process_form(submitted_form)
-        self._invalidate_caches(submitted_form)
+        context = form_submission_context_class()(self)
+        with context:
+            self.instance = context.get_instance() or self.instance
```

Comment:
rather have context always return an instance, even if it's unchanged

Reply:
thanks for catching that. context.get_instance should be enough and it already returns instance. I confused it with `context.get_instance_xml` and tried handling when xml fails to parse.
Updating this.
And handled it correctly in https://github.com/dimagi/commcare-icds/pull/41/commits/bd2a3c4a9684c96c1275f54afa12f59f4bb61560

Reply:
fixed in https://github.com/dimagi/commcare-hq/pull/28409/commits/2b6ecbddd5e9f46b3cbe9051c3242f1743e0b1fa